src/AppInstallerCommonCore/UserSettings.cpp

```diff
@@ -291,6 +294,24 @@ namespace AppInstaller::Settings
             return SettingMapping<Setting::InstallLocalePreference>::Validate(value);
         }
 
+        WINGET_VALIDATE_SIGNATURE(PortableAppUserRoot)
+        {
+            if (!std::filesystem::exists(value))
+            {
+                if (!std::filesystem::create_directories(value))

```

Comment:
Settings should not be creating directories.  Create a new `PathName` for `Runtime::GetPathTo`, retrieve the setting value and create the directory there.

Reply:
I created a new `PathName` for `PortableAppUserRoot`, `PortableAppMachineRootX64`, and `PortableAppMachineRootX86` and create the directories there after retrieving the setting value.